### DISCUSSION TITLE
error WIX0094: The identifier 'WixComponentGroup:XtraEditorsDemoCenter.Binaries' could not be found. Ensure you have typed the reference correctly and that all the necessary inputs are provided to the linker.

### DISCUSSION THREAD

**Discussion by haiconrbi** [wixtoolset/issues#7214] [Questions]
Which version of WiX are you building with?

    4.0.0-rc.2

    Which version of Visual Studio are you building with (if any)?

    2022

    Which version of the WiX Toolset Visual Studio Extension are you building with (if any)?

    HeatWave 0.9.4

    Which version of .NET are you building with?

    .net6.0

    Describe the problem and the steps to reproduce it.

 I converted the WiX3 project to WiX4. This worked in WiX3. In WiX4 has error:

 42>E:\Work\Solution\XtraEditors\MergeModule.wxs(7): error WIX0094: The identifier 'WixComponentGroup:XtraEditorsDemoCenter.Binaries' could not be found. Ensure you have typed the reference correctly and that all the necessary inputs are provided to the linker.
  
How to port from version 3x to version 4x properly RefProjectOutputGroups in ProjectReference?

XtraEditor.wixproj

\<Project Sdk="WixToolset.Sdk/4.0.0-rc.2">
  \<PropertyGroup>
    \<OutputType>Module</OutputType>
    \<OutputNameSuffix>.XtraEditorsDemoCenter</OutputNameSuffix>
    \<EnableProjectHarvesting>True</EnableProjectHarvesting>
    \<SuppressValidation>true</SuppressValidation>
  \</PropertyGroup>
  \<ItemGroup>
    \<ProjectReference Include="..\XtraEditorsDemoCenter\XtraEditorsDemoCenter.csproj">
      \<Name>XtraEditorsDemoCenter</Name>
      \<Project>{487e2f48-e213-466a-8abe-7b6bbf66fb00}</Project>
      \<Private>True</Private>
      \<DoNotHarvest></DoNotHarvest>
      \<RefProjectOutputGroups>Binaries;Content;Satellites</RefProjectOutputGroups>
      \<RefTargetDir>DEMOCENTER</RefTargetDir>
    \</ProjectReference>
  \</ItemGroup>
\</Project>

MergeModule.wxs

\<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">
  \<Module Language="1033" Codepage="1251" Version="$(var.FileVersion)" Id="$(var.MainProduct).$(var.SubProduct)_$(var.FileVersion).XtraEditorsDemoCenter" Guid="{171E70A8-686B-4AE4-93C1-02645515F75C}" InstallerVersion="200">
    \<SummaryInformation Description="$(var.MainProduct).$(var.SubProduct)-$(var.FileVersion).DemoCenter" Keywords="msi,installer" Manufacturer="$(var.Manufacturer)" Codepage="1251" />
    \<ComponentGroupRef Id="XtraEditorsDemoCenter.Binaries" />
    \<ComponentGroupRef Id="XtraEditorsDemoCenter.Content" />
      \<Directory Id="DEMOCENTER" Name="." />
    \</Module>
\</Wix>

**Comment by robmen** [comment]
What happens if you add a `PackageReference` to `WixToolset.Heat`?

**Comment by haiconrbi** [comment]
Could you describe in more detail. So far I don't quite understand your idea.

**Comment by haiconrbi** [comment]
I used [this artical](https://wixtoolset.org/docs/development/wips/organized-obsolecscence-of-heat-in-v4/) .

But unable to find package WixToolset.Heat 

```
nuget.exe install WixToolset.Heat.
Feeds used:                                             
https://api.nuget.org/v3/index.json
Installing package WixToolset.Heat to E:\.
CACHE https://api.nuget.org/v3/registration5-gz-semver2/wixtoolset.heat/index.json
Unable to find package WixToolset.Heat 
```

**Comment by robmen** [comment]
You can use HeatWave to add the package reference or edit your .wixproj to add:

```
<ItemGroup>
   <PackageReference Include="WixToolset.Heat" Version="4.0.0-rc.2" />
</ItemGroup>
```

**Comment by haiconrbi** [comment]
I already used it, same error

**Comment by robmen** [comment]
I don't know what "it" refers to in your statement. At this point, it's probably best if you provide a GitHub repo that demonstrates the issue.

**Comment by haiconrbi** [comment]
I created [repo](https://github.com/haiconrbi/Wix4Issue)

**Reply by robmen** [reply]
That repo is empty.

**Reply by haiconrbi** [reply]
Sorry, try check again

**Comment by barnson** [comment]
Converting to discussion.

**Comment by BMurri** [comment]
Three issues with the contents of the repo:

1. `EnableProjectHarvesting` needs to be set to `True` in the `PropertyGroup`.
2. `ReferenceOutputAssembly` should be False (it defaults to `True`, neither `heat` nor `wix build` use the effect of that being true, and it confuses nuget quite badly, especially when everything is using `PackageReference` and not the old-school `packages.config`).
3. You never define the directory `ClassLibrary1` (which comes from `RefTargetDir`).

Hope this helps someone in the future.
